src/custom/state/operator/middleware.ts

```diff
@@ -0,0 +1,46 @@
+import { Middleware } from '@reduxjs/toolkit'
+import { getTip } from '@src/custom/utils/fees'
+import { selectCurrency } from '@src/state/swap/actions'
+import { updateTip } from './actions'
+import { OperatorState } from './reducer'
+
+export const applyTipMiddleware: Middleware = ({ dispatch, getState }) => next => async action => {
+  // check that incoming action is a selected token
+  const isTriggerAction = action.type === selectCurrency.type
+
+  if (isTriggerAction) {
+    const { payload } = action
+
+    // check actions as several should trigger this update
+    // e.g direct selection? yes, also has payload to check token type
+    // but switchCurrencies? has no payload, but must be used to check tip
```

Comment:
Hey, wanna update **redux/toolkit** to latest to use [isAnyOf](https://github.com/reduxjs/redux-toolkit/pull/788) matchers?
then it would be `if(isAnyOf(selectCurrency, other actions that should trigger)(action))`

Reply:
:+1: